capstone/src/main/java/com/google/step/search/RestaurantQueryParams.java

```diff
@@ -0,0 +1,81 @@
+package com.google.step.search;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class to hold data for restaurant search queries.
+ */
+public class RestaurantQueryParams {
+  private final String query;
+  private final List<String> cuisine;
+  private final String email;
+
+  public static class Builder {
+    private String query = "";
+    private List<String> cuisine = new ArrayList<>();
+    private String email = "";
+
+    public Builder() { }
+
+    /**
+     * Sets query parameter to {@code query}
+     * @param query String representing a search query
+     * @return {@code this} builder
+     */
+    public Builder setQuery(String query) {
+      this.query = query;
+      return this;
+    }
+
+    /**
+     * Set cuisine to be {@code cuisineList}. The cuisine represents a list of cuisine types to be
+     * searched over.
+     * @param cuisineList list of strings representing cuisines
+     * @return {@code this} builder
+     */
+    public Builder setCuisine(List<String> cuisineList) {
```

Comment:
Just to keep this simple, I'd say pick one of two ways to set cuisine:

```
public Builder cuisine(List<String> cuisine)
or
public Builder cuisine(String... cuisine)
```

When you have both a set and an add, while it might be convenient, it also provides more than one (already-convenient) way to do it, and that can lead to maintenance burdens in the future, were this something you'd be developing over the years.

Reply:
I changed it to only take a list.